tests/nodes/tests_DBpediaMode.cpp

```diff
@@ -0,0 +1,60 @@
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
+#include <doctest/doctest.h>
+
+#include <rdf4cpp/rdf.hpp>
+
+using namespace rdf4cpp::rdf;
+
+TEST_CASE("IRI") {
+    IRIFactory f{};
+    static constexpr std::string_view iri = "http//:example:foo/bar";
+
+    datatypes::registry::dbpedia_mode = false;
+    CHECK(!f.create_and_validate(iri).has_value());
+
+    datatypes::registry::dbpedia_mode = true;
+    CHECK(f.create_and_validate(iri).has_value());
+    CHECK(f.create_and_validate(iri).value().identifier() == iri);
+}
+
+TEST_CASE("integers") {
+    Literal u{};
+    datatypes::registry::dbpedia_mode = false;
+    CHECK_THROWS_AS(u = Literal::make_typed<datatypes::xsd::NonNegativeInteger>("-5"), std::runtime_error);
+    CHECK_THROWS_AS(u = Literal::make_typed<datatypes::xsd::PositiveInteger>("0"), std::runtime_error);
+    CHECK(u == Literal{});
+
+    datatypes::registry::dbpedia_mode = true;
+    CHECK(Literal::make_typed<datatypes::xsd::NonNegativeInteger>("-5") == Literal::make_typed<datatypes::xsd::NonNegativeInteger>("0"));
```

Comment:
This behaviour is problematic because it changes the existing data. It would be better to just discard it.

Even better would be to make it into an `xsd:integer` and keep the value, but as we already discussed this is a little more complicated so I will create an issue for that specific approach, no need to implement it here.